### PR TITLE
Foreman Realm create/delete 

### PR DESCRIPTION
**Pull request by claudiol**
foreman-ansible-modules:
  - New Realm module to support create and delete of realms in Foreman/Satellite 6 modules/foreman_realm.py
  - Added realm.json file to support testing using Ansible hacking: ./.tmp/ansible/hacking/test-module -m modules/foreman_realm.py  -a @test/data/realm.json" test/data/realm.json
  - Added realm.yml as example on how to create/delete a realm" test/demo/realm.yml
  - Added choices to realm_type: "Red Hat Identity Management", "Active Directory", "FreeIPA"

Here is a test for deleting on a REALM:

```
TEST 1:
[claudiol@fedora26 lmco-cos]$ ansible-playbook -i ./inventory test-realm-module.yml

PLAY [satellite.redhat.local] **************************************************

TASK [Gathering Facts] *********************************************************
ok: [satellite.redhat.local]

TASK [Update REALM for an Organization] ****************************************
changed: [satellite.redhat.local]

TASK [debug] *******************************************************************
ok: [satellite.redhat.local] => {
    "response": {
        "changed": true,
        "failed": false
    }
}

PLAY RECAP *********************************************************************
satellite.redhat.local     : ok=3    changed=1    unreachable=0    failed=0  
```

TEST 2:
Here's a test creating a REALM:
```
[claudiol@fedora26 lmco-cos]$ ansible-playbook -i ./inventory test-realm-module.yml

PLAY [satellite.redhat.local] **************************************************

TASK [Gathering Facts] *********************************************************
ok: [satellite.redhat.local]

TASK [Update REALM for an Organization] ****************************************
changed: [satellite.redhat.local]

TASK [debug] *******************************************************************
ok: [satellite.redhat.local] => {
    "response": {
        "changed": true,
        "failed": false
    }
}

PLAY RECAP *********************************************************************
satellite.redhat.local     : ok=3    changed=1    unreachable=0    failed=0  
```

### CODE REVIEW

File: test/data/realm.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "server_url": "https://satellite.redhat.local",
+  "username": "admin",
+  "password": "redhat",
+  "verify_ssl": false,
+  "name": "EXAMPLE.COM",
+  "realm_proxy": 1,
+  "realm_type": "Red Hat Identity Management",
+  "state": present,
```
Comment (evgeni) [inline test/data/realm.json]: this should be quoted to be valid json, no?

**Comment by evgeni** [comment]
I did not test it yet (it requires https://github.com/SatelliteQE/nailgun/pull/449 and I was too lazy to patch my local nailgun with that yet), but the code looks pretty much correct to me.

Nice work!

**Comment by evgeni** [comment]
@akofink I think you forgot to patch your nailgun

File: test/data/realm.json (hunk 1 above)
Comment (claudiol) [reply test/data/realm.json]: Fixed typo in   695d168..8113741  claudiol-org-realm-update -> claudiol-org-realm-update
 push.

File: modules/foreman_realm.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+'''
+
+EXAMPLES = '''
+- name: "Create EXAMPLE.LOCAL Realm"
+  foreman_realm:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "EXAMPLE.COM"
+    real_proxy: 1
```
[the remark this answers is not in the document]
Comment (claudiol) [reply modules/foreman_realm.py]: Fixed typo in   695d168..8113741  claudiol-org-realm-update -> claudiol-org-realm-update.  That should do it for this.

**Pull request merged**